### DISCUSSION TITLE
[Bug]: Message.syncWithMeili and Message.syncWithMeili  "not a function"

### DISCUSSION THREAD

**Discussion by feeloow** [danny-avila/LibreChat#1506] [Troubleshooting] [answered]
### What happened?

On Ubuntu 20.04.6, manual installation, latest version, nothing dockerised.

Meilisearch running production with master key
Configuration OK
Search Engine displayed in Librechat  sidebar
No results when searching, API error

Startup Logs : 
`2024-01-07 12:15:35 error: [indexSync] error Message.syncWithMeili is not a function
`

Search Error Log
`error: [/search] Error while searching messages & conversations Message.meiliSearch is not a function`

API Response for GET GET /api/search?q=Shop&pageNumber=1'
`{"message":"Error searching"}`

### Steps to Reproduce

Install Librechat and Meilisearch manually 
Launch meiliseach with masterkey
Configure search for Librechat
Search any keyword from sidebar

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Can you share the logs in /api/logs concerning this? Likely a connection error to meilisearch

**Comment by wahitftry** [comment]
Hi. I have same problem. When trying to start docker container, the server throws an error: `[indexSync] Trouble creating indices, try restarting the server. Message.syncWithMeili is not a function`. Its always showed up each the docker container restarted

Here the message that i got from /api/logs/error-2024-01-07.log: 

`{"level":"error","message":"[indexSync] Trouble creating indices, try restarting the server. Message.syncWithMeili is not a function","stack":"TypeError: Message.syncWithMeili is not a function\n    at Timeout._onTimeout (/app/api/lib/db/indexSync.js:62:25)\n    at listOnTimeout (node:internal/timers:569:17)\n    at process.processTimers (node:internal/timers:512:7)"}`

**Comment by danny-avila** [comment, marked as answer]
Thanks for reporting this. Adding back to issues as I've identified what was causing it. 

For some reason, mongoose's execution order is starting before the `.env` file is loaded. There was nothing that should've changed this, but adding more operations at server initialization may have had this unintended effect on the call stack.

Adding `require('dotenv').config();` to top of /server/index.js fixes this
